### DISCUSSION TITLE
Object support for `fields` params

### DISCUSSION THREAD

**Discussion by kheiner** [Intevel/nuxt-directus#272] [General]
The `fields` param is not transformed on any request requiring the user to pre-format the strings within the array.  The `filter` and `deep` params are passed as objects and addressed through `JSON.stringify()` leading to the assumption that you can pass an objects into the `fields` array.

https://github.com/Intevel/nuxt-directus/blob/067b24331bfd558ae43fb683537a3be3ea2577f1/src/runtime/composables/useDirectusItems.ts#L18-L23

The directus/sdk handles this with a [util](https://github.com/directus/directus/blob/bb8bcbac9ab66efd97ca0601a55e8050f2c966d9/sdk/src/rest/utils/query-to-params.ts#L8-L42) that runs through the array to build the string.

Does it make sense to copy a similar util for this project or better to provide more details about how `fields` must be formatted in a unique way similar but not exactly the same as found in the directus docs for REST?

https://github.com/Intevel/nuxt-directus/blob/067b24331bfd558ae43fb683537a3be3ea2577f1/docs/content/2.composables/8.useDirectusItems.md?plain=1#L9

**Comment by casualmatt** [comment]
Hi @kheiner,

Thank you for your comment! As mentioned in discussion #271, we plan to rewrite the entire module in the coming months. If you have any suggestions, feedback, or feature requests, please share them there. This way, the contributors working on the new module can easily access all the relevant information in one thread.
